### ISSUE TITLE
Massive lag spike every 3 seconds and one crash.

### ISSUE THREAD

**Issue by jk1895** [vallantv/MBB#18]
I know you think that is not possible with such a simple modification. But it is. Every time I have your mod active I get lag after a certain time. When I deactivate it I don't get lag no matter how long I play.
Just now the game crashed when I clicked on your icon. The first time.

**Comment by rwilliams** [comment]
I just pushed a fix that we implemented on the Classic version of the addon. Let me know if you still have the issues. it's version 4.0.7

**Comment by rwilliams** [comment]
Can you also let me know what addons you're running?

**Comment by vallantv** [comment]
@jk1895 Try the latest version and let me know how it works. Can you give me a list of addons you're using that have lots of minimap markers? like gathering addons etc?

Thanks

**Comment by jk1895** [comment]
Thanks for taking a look at this.
I've played the 4.0.7 version for two hours now and haven't had any lag so far. Great!
My addons that I use and which place an icon next to the map:
Details!
Rarity
Paste
SimulationCraft
Hekili
WeakAuras
Deadly Boss Mods
Ask Mr. Robot
Altoholic

I don't really use addons that use map markers a lot. Maybe Wholly, HandyNotes for treasures and rares.
Other possible critical addons: FasterLoot, MoveAnything

My guess is that Rarity is the culprit. When you mouse over the icon of this addon, the game hangs for two seconds. Maybe this is somehow related.
By the way, I tried MBB Reborn and it lagged the same way.

Thank you.
